prow/README.md

```diff
@@ -4,206 +4,383 @@ Metal3 Prow dashboard: <https://prow.apps.test.metal3.io>
+- [openstack](https://docs.openstack.org/newton/user-guide/common/cli-install-openstack-command-line-clients.html),
+  can be installed directly from Ubuntu repositories.
+- [s3cmd](https://s3tools.org/s3cmd), can be installed directly from Ubuntu repositories.
+
+### Folders and components
+
+There are three folders with kustomizations (`capo-cluster`, `infra` and `manifests`).
+The `capo-cluster` folder contains everything needed for creating the Kubernetes
+cluster itself.
+In `infra`, you will find the necessary parts for integrating with the
+"infrastructure", i.e. the external cloud-provider for OpenStack, a ClusterIssuer
+for getting Let's Encrypt certificates, a LoadBalancer Service for ingress and a
+StorageClass for Cinder volumes.
+
+The deployment manifests for Prow (`manifests`) are based on the
+[getting started guide](https://docs.prow.k8s.io/docs/getting-started-deploy/)
+and the
+[starter-s3.yaml](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/starter/starter-s3.yaml)
+manifests.
+They have been separated out into multiple files and tied together using kustomizations.
+This makes it easy to keep secrets outside of git while still allowing for simple
+one-liners for applying all of it.
+
+The getting started guide is (as of writing) focused on setting up Prow using a
+GitHub app, but Prow actually supports [many authentication methods](https://github.com/kubernetes/test-infra/blob/441b5000ef79491d08bd5f393de8f69650b85355/prow/flagutil/github.go#L137).
+We use API tokens created from GitHub bot accounts.
+
+We use [ingress-nginx](https://kubernetes.github.io/ingress-nginx/) as ingress controller.
+It is fronted by a LoadBalancer Service, i.e. a loadbalancer in Cleura.
+The Service is configured to grab the correct IP automatically and to avoid
+deleting it even if the Service is deleted.
+See [infra/service.yaml](infra/service.yaml).
+For securing it with TLS we rely on [cert-manager](https://cert-manager.io/) and
+the Let's Encrypt HTTP01 challenge, as seen in [infra/cluster-issuer-http.yaml](infra/cluster-issuer-http.yaml).
+
+## Building node images
+
+The Kubernetes cluster where Prow runs needs pre-built images for the Nodes.
+We use [image-builder](https://github.com/kubernetes-sigs/image-builder) for this.
+
+Here is how to build a node image directly in Cleura.
+See the [image-builder book](https://image-builder.sigs.k8s.io/capi/providers/openstack-remote.html)
+for more details.
+Start by creating a JSON file with relevant parameters for the image.
+Here is an example:
+
+```json
+{
+   "source_image": "d6b012ee-c6d3-4672-9399-b87d025ddb14",
+   "networks": "bda060e0-07a2-4ba2-8fe0-dba7626db240",
+   "flavor": "4C-4GB-20GB",
+   "floating_ip_network": "ext-net",
+   "ssh_username": "ubuntu",
+   "volume_type": "",
+   "kubernetes_deb_version": "1.26.3-00",
+   "kubernetes_rpm_version": "1.26.3-0",
+   "kubernetes_semver": "v1.26.3",
+   "kubernetes_series": "v1.26"
+}
```

Comment:
so uplifting k8s version would mean creating node images and doing rolling upgrades? The images stay in openstack only right?